src/libraries/System.Net.Http/src/System/Net/Http/MetricsHandler.cs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Net.Http.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http;
+
+internal sealed class MetricsHandler : HttpMessageHandlerStage, IHttpMetricsLogger
+{
+    private readonly HttpMessageHandler _innerHandler;
+    private readonly UpDownCounter<long> _currentRequests;
+    private readonly Counter<long> _failedRequests;
+    private readonly Histogram<double> _requestsDuration;
+
+    private ThreadLocal<HttpMetricsEnrichmentContext> _cachedEnrichmentContext = new ThreadLocal<HttpMetricsEnrichmentContext>(() => new HttpMetricsEnrichmentContext());
+
+    public MetricsHandler(HttpMessageHandler innerHandler, Meter meter)
+    {
+        _innerHandler = innerHandler;
+
+        if (meter is SharedMeter sharedMeter)
+        {
+            Debug.Assert(ReferenceEquals(meter, DefaultMeter));
+            _currentRequests = sharedMeter._currentRequests;
+            _failedRequests = sharedMeter._failedRequests;
+            _requestsDuration = sharedMeter._requestsDuration;
+        }
+        else
+        {
+            CreateInstruments(meter, out _currentRequests, out _failedRequests, out _requestsDuration);
+        }
+    }
+
+    public static Meter DefaultMeter { get; } = new SharedMeter();
```

Comment:
```suggestion
    public static readonly Meter DefaultMeter { get; } = new SharedMeter();
```

Reply:
This is a property in a class, you can't apply readonly here AFAIK.